pgx-utils/src/lib.rs

```diff
@@ -408,6 +408,22 @@ pub fn staticize_lifetimes(value: &mut syn::Type) {
             }
         }
 
+        syn::Type::Macro(type_macro) => {
+            let mac = &type_macro.mac;
+            let archetype = mac.path.segments.last().unwrap();
+            match archetype.ident.to_string().as_str() {
+                "name" => {
+                    let mut out: NameMacro = mac.parse_body().expect("name!() in wrong format");
+                    staticize_lifetimes(&mut out.used_ty.resolved_ty);
+
+                    // rewrite the name!() macro so that it has a static lifetime, if any
+                    let ident = syn::parse_str::<Ident>(&out.ident).unwrap();
+                    let ty = out.used_ty.resolved_ty;
+                    type_macro.mac = syn::parse_quote! {name!(#ident, #ty)};
```

Comment:
```rust
#[pg_extern]
fn split_table_with_borrow<'a>(
    input: &'a str,
    pattern: &'a str,
) -> TableIterator<'a, (name!(i, i32), name!(s, &'a str))>
```
So in this case, the return lifetimes we are expecting will be
```rust
) -> TableIterator<'static, (name!(i, i32), name!(s, &'static str))>
```
?

Reply:
Hm. I think this can be done in a better way in the future, but I can't find the solution immediately and we're staticizing other lifetimes, so I think this is acceptable for the current moment.